code/modules/library/book.dm

```diff
@@ -0,0 +1,370 @@
+  *
+  * Gives free pass to observers to read, ensures that all other mobs attempting to read book are A) literated and
+  * B) are within range to actually read the book.
+  */
+/obj/item/book/proc/can_read(mob/user)
+	if(isobserver(user)) //We check this first because ghosts should be able to read any book they can see no matter what
+		return TRUE
+	if(!in_range(src, user))
+		return FALSE
+	if(!user.is_literate()) //this person was 2 cool 4 school and cannot READ
+		user.visible_message("You attempt to the read the book but remember that you don't actually know how to read.")
+		return FALSE
+	return TRUE
+
+/**
+  * Read Book Proc
+  *
+  * Checks if players is able to read book and that book is readable before calling the neccesary procs to open up UI
+  */
+/obj/item/book/proc/read_book(mob/user)
+	if(!length(pages)) //You can't read a book with no pages in it
+		to_chat(user, "This book is completely blank!")
+		return
+	if(!can_read(user))
+		return
+	user.visible_message("[user] opens a book titled \"[title]\" and begins reading intently.")
+	show_content(user) //where all the magic happens
+	onclose(user, "book")
+
+/**
+  * Show Content Proc
+  *
+  * Builds the browserUI html to show to the player then open up the User interface. It first build the header navigation
+  * buttons and then builds the rest of the UI based on what page the player is turned to.
+  */
+/obj/item/book/proc/show_content(mob/user)
+	var/dat = " "
+	//First, we're going to choose/generate our header buttons for switching pages and store it in var/dat
+	if(length(pages)) //No need to have page switching buttons if there's no pages
+		if(current_page == 0)
+			dat+= "<DIV STYLE='float:left; text-align:left; width:49.9%'></DIV>"
+			dat+= "<DIV STYLE='float:left; text-align:right; width:49.9%'><A href='?src=[UID()];next_page=1'>Next Page</A></DIV><BR><HR>"
+		else if(current_page < length(pages))
+			dat+= "<DIV STYLE='float:left; text-align:left; width:49.9%'><A href='?src=[UID()];prev_page=1'>Previous Page</A></DIV>"
+			dat+= "<DIV STYLE='float:left; text-align:right; width:49.9%'><A href='?src=[UID()];next_page=1'>Next Page</A></DIV><BR><HR>"
+		else if(current_page == length(pages))
+			dat+= "<DIV STYLE='float:left; text-align:left; width:49.9%'><A href='?src=[UID()];prev_page=1'>Previous Page</A></DIV>"
+			dat+= "<DIV STYLE='float;left; text-align:right; width:49.9%'></DIV>"
+
+	//Now we're going to display the header buttons + the current page selected, if it's page 0, we display the cover_page instead
+	if(!current_page)
+		var/cover_page = {"<center><h1>[title]</h1><br></h2>Written by: [author]</h2></center><br><hr><b>Summary:</b> [summary]"}
+		user << browse("<body bgcolor='[book_bgcolor]'>[dat]<BR>" + "[cover_page]", "window=book[UID()];size=400x400")
+		return
+	else
+		user << browse("<body bgcolor='[book_bgcolor]'>[dat]<BR>" + "[pages[current_page]]", "window=book[UID()]")
+
+/obj/item/book/Topic(href, href_list)
+	..()
```

Comment:
Prob best to make use of the return value here. A lot of things are already checked for you if somebody can use the item or not in `/obj/Topic`

Reply:
should do that now